lib/data/external/GCP/GcpApis/mpuHelper.js

```diff
@@ -0,0 +1,362 @@
+     * @return {undefined}
+     */
+    removeParts(params, callback) {
+        const _getObjectVersions = (bucketType, callback) => {
+            logger.trace(`remove all parts ${bucketType} bucket`);
+            let partList = [];
+            let isTruncated = true;
+            let nextMarker;
+            const bucket = params[bucketType];
+            return async.whilst(() => isTruncated, next => {
+                const listParams = {
+                    Bucket: bucket,
+                    Prefix: params.Prefix,
+                    Marker: nextMarker,
+                };
+                return this.service.listVersions(listParams, (err, res) => {
+                    if (err) {
+                        logHelper(logger, 'error', 'error in ' +
+                            `removeParts - listVersions ${bucketType}`, err);
+                        return next(err);
+                    }
+                    nextMarker = res.NextMarker;
+                    isTruncated = res.IsTruncated;
+                    partList = partList.concat(res.Versions);
+                    return next();
+                });
+            }, err => callback(err, partList));
+        };
+
+        const _deleteObjects = (bucketType, partsList, callback) => {
+            logger.trace(`successfully listed ${bucketType} parts`, {
+                objectCount: partsList.length,
+            });
+            const delSlices = this.createDelSlices(partsList);
+            const bucket = params[bucketType];
+            return async.each(delSlices, (list, next) => {
+                const delParams = {
+                    Bucket: bucket,
+                    Delete: { Objects: list },
+                };
+                return this.service.deleteObjects(delParams, err => {
+                    if (err) {
+                        logHelper(logger, 'error',
+                            `error deleting ${bucketType} object`, err);
+                    }
+                    return next(err);
+                });
+            }, err => callback(err));
+        };
+
+        return async.parallel([
+            done => async.waterfall([
+                next => _getObjectVersions('MPU', next),
+                (parts, next) => _deleteObjects('MPU', parts, next),
+            ], err => done(err)),
+            done => async.waterfall([
+                next => _getObjectVersions('Overflow', next),
+                (parts, next) => _deleteObjects('Overflow', parts, next),
+            ], err => done(err)),
+        ], err => callback(err));
+    }
+
+    copyToOverflow(numParts, params, callback) {
+        // copy phase: in overflow bucket
+        // resetting component count by moving item between
+        // different region/class buckets
+        logger.trace('completeMultipartUpload: copy to overflow',
+            { partCount: numParts });
+        const parts = createMpuList(params, 'mpu2', numParts);
+        let doneCount = 0;
```

Comment:
I'm not sure I understand this, is it just for the next function in the waterfall [here](https://github.com/scality/S3/pull/1101/files#diff-fd2f8a83c6db0697739f6451b27c2317R67)? If so, I think it would be better to pass `numParts` to the next step in the waterfall rather than increment this and return it in the callback.

Reply:
`numParts` can be passed the callback. Had made changes to add the helper, and that was left in.